appserver/admingui/devtests/run_test.sh

```diff
@@ -0,0 +1,106 @@
+#
+# GPL Classpath Exception:
+# Oracle designates this particular file as subject to the "Classpath"
+# exception as provided by Oracle in the GPL Version 2 section of the License
+# file that accompanied this code.
+#
+# Modifications:
+# If applicable, add the following below the License Header, with the fields
+# enclosed by brackets [] replaced by your own identifying information:
+# "Portions Copyright [year] [name of copyright owner]"
+#
+# Contributor(s):
+# If you wish your version of this file to be governed by only the CDDL or
+# only the GPL Version 2, indicate your decision by adding "[Contributor]
+# elects to include this software in this distribution under the [CDDL or GPL
+# Version 2] license."  If you don't indicate a single choice of license, a
+# recipient has the option to distribute your version of this file under
+# either the CDDL, the GPL Version 2 or to extend the choice of license to
+# its licensees as provided above.  However, if you add GPL Version 2 code
+# and therefore, elected the GPL Version 2 license, then the option applies
+# only if the new code is made subject to such option by the copyright
+# holder.
+
+list_test_ids(){
+  echo admingui_all
+}
+
+merge_junit_xmls(){
+  JUD_DIR=$1
+  JUD=$WORKSPACE/results/junitreports/test_results_junit.xml
+  rm -f ${JUD} || true
+  echo "<?xml version=\"1.0\" encoding=\"UTF-8\"?>" >> ${JUD}
+  echo "<testsuites>" >> ${JUD}
+  find ${JUD_DIR} -name "*.xml" -type f -exec cat '{}' \; | ${SED} 's/<?xml version=\"1.0\" encoding=\"UTF-8\" ?>//g' >> ${JUD}
+  echo -e "\n</testsuites>" >> ${JUD}
+}
+
+test_run(){
+  touch /tmp/password.txt
+  chmod 600 /tmp/password.txt
+  echo "AS_ADMIN_PASSWORD=" > /tmp/password.txt
+  echo "AS_ADMIN_NEWPASSWORD=admin" >> /tmp/password.txt
+  $S1AS_HOME/bin/asadmin --user admin --passwordfile /tmp/password.txt change-admin-password
+  $S1AS_HOME/bin/asadmin start-domain
+  echo "AS_ADMIN_PASSWORD=admin" > /tmp/password.txt
+  $S1AS_HOME/bin/asadmin --passwordfile /tmp/password.txt enable-secure-admin
+  $S1AS_HOME/bin/asadmin restart-domain
+  cd $APS_HOME/../../admingui/devtests/
+  export DISPLAY=127.0.0.1:1	
+  mvn -Dmaven.repo.local=$WORKSPACE/repository -DsecureAdmin=true test | tee $TEST_RUN_LOG
+  $S1AS_HOME/bin/asadmin stop-domain
+  rm -rf /tmp/password.txt	
+  cp $WORKSPACE/bundles/version-info.txt $WORKSPACE/results/
+  cp $TEST_RUN_LOG $WORKSPACE/results/
+  cp $WORKSPACE/glassfish5/glassfish/domains/domain1/logs/server.log* $WORKSPACE/results/ || true  
+}
+
+run_test_id(){
+  #a common util script located at main/appserver/tests/common_test.sh
+  source `dirname $0`/../../tests/common_test.sh
+  kill_process
+  delete_gf
+  export M2_HOME=/net/gf-hudson/scratch/gf-hudson/export2/hudson/tools/apache-maven-3.0.3
+  export MAVEN_OPTS="-Xmx1024m -XX:MaxPermSize=384m"
+  export MAVEN_REPO=$WORKSPACE/repository
+  export MAVEN_SETTINGS=$M2_HOME/settings-nexus.xml
+  export PATH=$M2_HOME/bin:$JAVA_HOME/bin:$PATH
```

Comment:
1. M2_HOME, PATH is already set in common_test.sh, no need to set here.
2. Did you mean to set M2_REPO above? MAVEN_REPO is not used.